pkgs/development/python-modules/spsdk/default.nix

```diff
@@ -0,0 +1,123 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchpatch
+, dos2unix
+, asn1crypto
+, astunparse
+, bincopy
+, bitstring
+, click
+, click-option-group
+, cmsis-pack-manager
+, commentjson
+, crcmod
+, cryptography
+, deepmerge
+, fastjsonschema
+, hexdump
+, jinja2
+, libusbsio
+, oscrypto
+, pycryptodome
+, pylink-square
+, pyocd
+, pypemicro
+, pyserial
+, ruamel-yaml
+, sly
+, pytestCheckHook
+, voluptuous
+, binutils
+, openssl
+}:
+
+buildPythonPackage rec {
+  pname = "spsdk";
+  version = "1.6.3";
+
+  # pypi does not include tests (https://github.com/NXPmicro/spsdk/issues/37)
+  src = fetchFromGitHub {
+    owner = "NXPmicro";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-JMhd2XdbjEN6SUzFgcBHd/dStiuYeXXis6pfijSfUso=";
+  };
+
+  patches = [
+    # https://github.com/NXPmicro/spsdk/pull/43
+    (fetchpatch {
+      name = "cryptography-37-compat.patch";
+      url = "https://github.com/NXPmicro/spsdk/commit/a85b854de1093de593d27fa64de442224ab2e0fd.patch";
+      sha256 = "sha256-4pXV/8RaNuGl7KNdoGD/8YnPQ2ZmUQOjXWA/Yy0Kxu8=";
+    })
+    # https://github.com/NXPmicro/spsdk/pull/41
+    (fetchpatch {
+      name = "blhost-click-8-1-compat.patch";
+      url = "https://github.com/NXPmicro/spsdk/commit/5112b1b69aa681d265035475e73d28ea0c8cb6ab.patch";
+      sha256 = "sha256-Okz6Er6OVuAA5IlB5IabSa/gUSLa+E2Ltd+J3uoIg6o=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace requirements.txt \
+        --replace "cmsis-pack-manager<0.3.0" "cmsis-pack-manager" \
+        --replace "cryptography>=3.4.4,<37" "cryptography" \
+        --replace "deepmerge==0.3.0" "deepmerge" \
+        --replace "jinja2>=2.11,<3.1" "jinja2" \
+        --replace "pylink-square>=0.8.2,<0.11.9" "pylink-square" \
+        --replace "pyocd-pemicro>=1.1.1,<1.2.0" "" \
+        --replace "pyocd>=0.28.3,<=0.31.0" "pyocd"
+  '';
+
+  propagatedBuildInputs = [
+    asn1crypto
+    astunparse
+    bincopy
+    binutils.bintools # required for ctypes to find libcrypto (https://github.com/dbt-labs/dbt-core/issues/3366#issuecomment-849764321)
```

Comment:
Can we patch the code instead?

Reply:
thats strange

Reply:
binutils (objdump) is indeed one of the methods for ctypes to find dependencies.

Still, in Nixpkgs we patch out ctypes calls in the calling libraries. This is not for the interpreter.